lucene/sandbox/src/test/org/apache/lucene/search/TestLargeNumHitsTopDocsCollector.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.search;
+
+import java.io.IOException;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.RandomIndexWriter;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.util.LuceneTestCase;
+
+public class TestLargeNumHitsTopDocsCollector extends LuceneTestCase {
+  private Directory dir;
+  private IndexReader reader;
+  private final Query testQuery = new BooleanQuery.Builder()
+      .add(new TermQuery(new Term("field", "5")), BooleanClause.Occur.SHOULD)
+      .add(new MatchAllDocsQuery(), BooleanClause.Occur.SHOULD)
+      .build();
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    dir = newDirectory();
+    RandomIndexWriter writer = new RandomIndexWriter(random(), dir);
+    for (int i = 0; i < 200_000; i++) {
+      if (random().nextBoolean()) {
+        Document doc = new Document();
+        doc.add(newStringField("field", "5", Field.Store.NO));
+        writer.addDocument(doc);
+      }
+      else {
```

Comment:
we usually put the else statement on the same line as the closing bracket of the if statement